mars/dataframe/indexing/getitem.py

```diff
@@ -0,0 +1,195 @@
+                                      chunks=out_chunks)
+
+    @classmethod
+    def execute(cls, ctx, op):
+        df = ctx[op.inputs[0].key]
+        ctx[op.outputs[0].key] = df[op.labels]
+
+
+class DataFrameIndex(DataFrameOperand, DataFrameOperandMixin):
+    _op_type_ = OperandDef.INDEX
+
+    _col_names = AnyField('col_names')
+
+    def __init__(self, col_names=None, object_type=ObjectType.series, **kw):
+        super(DataFrameIndex, self).__init__(_col_names=col_names, _object_type=object_type, **kw)
+
+    @property
+    def col_names(self):
+        return self._col_names
+
+    def __call__(self, df):
+        # if col_names is a tuple or list, return a DataFrame, else return a Series
+        if isinstance(self._col_names, (list, tuple)):
+            dtypes = df.dtypes[self._col_names]
+            columns = parse_index(pd.Index(self._col_names), store_data=True)
+            return self.new_dataframe([df], shape=(df.shape[0], len(self._col_names)), dtypes=dtypes,
+                                      index_value=df.index_value, columns_value=columns)
+        else:
+            dtype = df.dtypes[self._col_names]
+            return self.new_series([df], shape=(df.shape[0],), dtype=dtype, index_value=df.index_value,
+                                   name=self._col_names)
+
+    @classmethod
+    def tile(cls, op):
+        in_df = op.inputs[0]
+        out_df = op.outputs[0]
+        col_names = op.col_names
+        if not isinstance(col_names, (tuple, list)):
+            column_index = calc_columns_index(col_names, in_df)
+            out_chunks = []
+            dtype = in_df.dtypes[col_names]
+            for i in range(in_df.chunk_shape[0]):
+                c = in_df.cix[(i, column_index)]
+                op = DataFrameIndex(col_names=col_names)
+                out_chunks.append(op.new_chunk([c], shape=(c.shape[0],), index=(i,), dtype=dtype,
+                                               index_value=c.index_value, name=col_names))
+            new_op = op.copy()
+            return new_op.new_seriess(op.inputs, shape=out_df.shape, dtype=out_df.dtype,
+                                      index_value=out_df.index_value, name=out_df.name,
+                                      nsplits=(in_df.nsplits[0],), chunks=out_chunks)
+        else:
+            selected_index = []
+            for col_name in col_names:
+                column_index = calc_columns_index(col_name, in_df)
+                selected_index.append(column_index)
+
+            # combine columns in one chunk and keep the columns order at the same time.
+            column_splits = []
+            start = 0
+            for i in range(len(selected_index)):
+                if i == len(selected_index) - 1 or selected_index[i] != selected_index[i + 1]:
+                    column_splits.append((col_names[slice(start, i + 1)], selected_index[start]))
+                    start = i + 1
```

Comment:
This part looks really complicated... How to about grouping the neighboring column names like this:

```python
selected_index = [calc_columns_index(col, in_df) for col in col_names]
column_splits = np.split(col_names, np.where(np.diff(selected_index))[0] + 1)
column_indexes = np.split(selected_index, np.where(np.diff(selected_index))[0] + 1)
```